Marco Segura Coding Test

3birdsmedia opened the pull request:
**Component Created:** ~/components/gallery/list.vue from #3

**Stories:** stories/GlobalGalleryList.stories.js

**Imports:**

- "~/components/global/Logo";
- "~/components/global/Hamburger";
- "~/assets/svg/logo-rock-paper-scissors";
- "~/components/WpImage";
- "~/stories/mock-api.json";

**Notes:**

- I had to go back in the history folder to get a missing Funkhaus logo that was creating an issue on the first compile
- On hover the images are loaded dynamically to decrease page load time and bloat.
- Used CSS grid instead of Flexbox to accomplish a look closer to the design

**Time Report:**

This took me 5 hours to build this.

**Checklist:**

-   [x] I double checked it looks like the designs
-   [x] I completed any required mobile breakpoint styling
-   [x] I completed any required hover state styling
-   [x] I included a working Storybook file
-   [ ] I included a Story that showed some edge case working correctly (long text, short screen, missing image etc.)
-   [x] I added notes above about how long it took to build this component
-   [x] I assigned this PR to someone to review

3birdsmedia committed:
Test finished, had to add the funkhaus logo that was missing from a previous commit. Test took me approximatedly 1hr setup/read through, 3hrs code, 2hrs debbuging/cleaning code to funkhaus standards.

3birdsmedia committed:
Extra commit to run code through Prettier

drewbaker reviewed (changes requested):
Thanks for this @3birdsmedia. Here are some notes for you.

- [ ] We only need you to build the gallery-list component (as shown in the screenshots), not the entire page. So you don't need to the hamburger/logo/breadcrumb. 
- [ ] You should be able to config VS Code prettier to pull it's settings from the included ES lint config file. We don't want your settings.json file included in this PR.
- [ ] You have stray `console.log`. Please remove.
- [ ] `wp-image` can just work with `:image` prop in this case. _I probably need to document how wp-image works, so this isn't your fault._
- [ ] Why are you using `!important` that is a big red-flag.
- [ ] The font isn't styled correctly as per designs.
- [ ] The last column of names should line up much closer to the right edge as per designs.
- [ ] On mobile breakpoint the names should be centered in middle of screen
- [ ] Why the `z-index:0` on gallery-list at line 70?
- [ ] I encourage you to read about how (and why) `key` works in Vue. You are using it wrong.

3birdsmedia committed:
Made revisions necesary to component

3birdsmedia commented:
Thank you for your feedback @drewbaker I have revised the component, and pushed to master. See below from my explanation to some items that needed it. 

 - [x] We only need you to build the gallery-list component (as shown in the screenshots), not the entire page. So you don't need to the hamburger/logo/breadcrumb. `I have removed the unnecessary components`

- [x] You should be able to config VS Code prettier to pull it's settings from the included ES lint config file. We don't want your settings.json file included in this PR. `My apologies, it must have sneaked in when I was setting up the project to use the 4 space tabs, I have removed it from the commit`

- [x] You have stray console.log. Please remove. `Removed`

- [x] wp-image can just work with :image prop in this case. I probably need to document how wp-image works, so this isn't your fault. `No worries, I was able to use the image prop, and pass the image objects and it worked as intended (loading a video when present`

- [x] Why are you using !important that is a big red-flag. `I have removed it, it was a left over from when I was testing sizes`

- [x] The font isn't styled correctly as per designs. `I have changed the font to RM Neue at 24px, with 37px  padding as per the XD file specifications`

- [x] The last column of names should line up much closer to the right edge as per designs. `I have moved it to the right by 5%`

- [x] On mobile breakpoint the names should be centered in middle of screen `This has been implemented`

- [x] Why the z-index:0 on gallery-list at line 70? `On your paper documentation "How to build a component for Funkhaus", Section 9 CSS structure, point number 14, it mentions if needing to use z-index, to set the root element's index to 0 and build on 10s from there, I can remove it and the design will still work. I was trying to follow the docs :)`

- [x] I encourage you to read about how (and why) key works in Vue. You are using it wrong. `I see that I was adding it when it wasn't needed, I brought that piece of code in from the "shortcode/gallery.vue" component, in which key is being used.`

drewbaker commented:
Thanks Marco. We only do one round of review on these code tests so you don't need to do any more on this. But just for training purposes I'll list here:

1. You're right about`z-index` and the docs, but without `position: relative` it's doing nothing. You also need that.
2. The correct way to do the list of names is to use two flexbox's. That allows for no hard coding of offset values. I think it's impossible with grid. 
3. Your PR still includes the logo and your story has old code (menu event) in it that isn't being used now.
4. `wp-image` styling should be `position: fixed` so its centered on scroll, and the `transform: translate(-50%, -50%)` isn't doing anything. 
5. Bonus points if you'd made the directors list a `ul` with nested `<ul> and <li>` for columns.
6. Bonus points if the names where `<h3>` or another header element for SEO.
6. Our style is to avoid styling elements directly. So would prefer to have classes on the `a` elements.
7. We like to have the hover state in a media query at end of CSS [as shown here](https://paper.dropbox.com/doc/How-to-build-a-component-for-Funkhaus-0Q3kfQgX6NgLdBwkH8OyR#:h2=CSS-structure).
8. You image animation fades in, but turns off instantly. You'd need a transition style to solve this.
9. You are importing `WpImage` into your story for no reason.
